homeassistant/components/lcn/cover.py

```diff
@@ -66,26 +67,47 @@ async def async_added_to_hass(self):
     @property
     def is_closed(self):
         """Return if the cover is closed."""
-        return self._closed
+        return self._is_closed
+
+    @property
+    def is_opening(self):
+        """Return if the cover is opening or not."""
+        return self._is_opening
+
+    @property
+    def is_closing(self):
+        """Return if the cover is closing or not."""
+        return self._is_closing
+
+    @property
+    def assumed_state(self):
+        """Return True if unable to access real state of the entity."""
+        return True
 
     async def async_close_cover(self, **kwargs):
         """Close the cover."""
-        self._closed = True
-
+        self._is_closed = True
+        self._is_opening = False
+        self._is_closing = True
         state = pypck.lcn_defs.MotorStateModifier.DOWN
         self.address_connection.control_motors_outputs(state)
         self.async_write_ha_state()
 
     async def async_open_cover(self, **kwargs):
         """Open the cover."""
-        self._closed = False
+        self._is_closed = False
+        self._is_opening = True
+        self._is_closing = False
         state = pypck.lcn_defs.MotorStateModifier.UP
         self.address_connection.control_motors_outputs(state, self.reverse_time)
         self.async_write_ha_state()
 
     async def async_stop_cover(self, **kwargs):
         """Stop the cover."""
-        self._closed = None
+        if self._is_opening or self._is_closing:
+            self._is_closed = self._is_closing
```

Comment:
Should we really assume that the cover was stopped in a closed position if it was closing? Isn't it more probably that the cover was stopped before reaching the closed position? Why would we stop it when it's already at an endpoint?

Reply:
That is a question of definition I was unsure about. If we stop the cover at 50%, is it half open or half closed? Depending on that the frontend will show a full open or full closed icon when the cover stops. Shall I change it, so that the cover is assumed to be open in case it is not fully closed?

Reply:
Yes, the cover is always open until it's fully closed.